### PR TITLE
Merge paradox/scala/camel.md and java/camel.md

### PR DESCRIPTION
**Pull request by richardimaoka**
**Ready for review.**

Below cla valiator is because of my GitHub ID changed from `richard-imaoka` to `richardimaoka` recently. I signed up again already.

This is another pull request with messy code diff. As far as I compared the rendered HTML with the ones in the current ver, I saw them consistent, except for intended changes as I commented in this PR.

Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/camel.html
- https://richard-akka-doc.firebaseapp.com/java/camel.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/camel.md and java/camel.md -> **this is to take the diff both in scala/camel.md and java/camel.md**

  - [scala/camel.md  (prev) vs. scala/camel.md (after)](https://github.com/akka/akka/pull/23456/files#diff-d25d4967f5aa29399dfcf5117c1a2aee)
  - [java/camel.md (prev) vs. java/camel.md (after)](https://github.com/akka/akka/pull/23456/files#diff-0c6664152eac29b45a037b0801c77ad9)

- **2nd commit:** Replace java/camel.md with a symlink to scala/camel.md

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @richardimaoka,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: akka-docs/src/main/paradox/scala/camel.md [hunk 1]
```diff
@@ -24,31 +24,57 @@ APIs. The [camel-extra](http://code.google.com/p/camel-extra/) project provides
...
+Camel's [Mina component](http://camel.apache.org/mina2.html). The actor implements the @scala[`endpointUri`]@java[`getEndpointUri`] method to define
 an endpoint from which it can receive messages. After starting the actor, TCP
 clients can immediately send messages to and receive responses from that
 actor. If the message exchange should go over HTTP (via Camel's Jetty
-component, only the actor's endpointUri method must be changed.
+component), the actor's @scala[`endpointUri`]@java[`getEndpointUri`] method should return a different URI, for instance `jetty:http://localhost:8877/example`.
+
+@@@ div { .group-scala }
 
 @@snip [Introduction.scala]($code$/scala/docs/camel/Introduction.scala) { #Consumer }
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: No equivalent in Java

File: akka-docs/src/main/paradox/scala/camel.md [hunk 2]
```diff
@@ -24,31 +24,57 @@ APIs. The [camel-extra](http://code.google.com/p/camel-extra/) project provides
...
 
+@@@ 
+
+@@@ div { .group-java } 
+
+In the above case an extra constructor is added that can set the endpoint URI, which would result in
+the `getEndpointUri` returning the URI that was set using this constructor.
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: Extra constructor is specific to [Java](http://doc.akka.io/docs/akka/current/java/camel.html#consumer) (not relevant in [Scala](http://doc.akka.io/docs/akka/current/scala/camel.html#consumer)).

```java
  // Extra constructor to change the default uri,
  // for instance to "jetty:http://localhost:8877/example"
  public MyEndpoint(String uri) {
    this.uri = uri;
  }
```

File: akka-docs/src/main/paradox/scala/camel.md [hunk 3]
```diff
@@ -24,31 +24,57 @@ APIs. The [camel-extra](http://code.google.com/p/camel-extra/) project provides
...
 In the above example, any message sent to this actor will be sent to
-the JMS queue `orders`. Producer actors may choose from the same set of Camel
+the JMS queue @scala[`orders`]@java[`Orders`]. Producer actors may choose from the same set of Camel
 components as Consumer actors do.
 
+@@@ div { .group-java } 
+
+Below an example of how to send a message to the `Orders` producer.
+
+@@snip [ProducerTestBase.java]($code$/java/jdocs/camel/ProducerTestBase.java) { #TellProducer }
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: No equivalent code snippet for Scala.

File: akka-docs/src/main/paradox/scala/camel.md [hunk 4]
```diff
@@ -69,68 +95,93 @@ You can also create a CamelMessage yourself with the appropriate body and header
...
+:  @@snip [Consumers.scala]($code$/scala/docs/camel/Consumers.scala) { #Consumer1 }
+
+Java
+:  @@snip [Consumer1.java]($code$/java/jdocs/camel/Consumer1.java) { #Consumer1 }
 
 Whenever a file is put into the data/input/actor directory, its content is
 picked up by the Camel [file component](http://camel.apache.org/file2.html) and sent as message to the
 actor. Messages consumed by actors from Camel endpoints are of type
-@extref[CamelMessage](github:akka-camel/src/main/scala/akka/camel/CamelMessage.scala). These are immutable representations of Camel messages.
+[CamelMessage](#camelmessage). These are immutable representations of Camel messages.
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: Think it's better to link to a section in the document, as it has more explanation than the code. (in line with Java)

File: akka-docs/src/main/paradox/scala/camel.md [hunk 5]
```diff
@@ -169,73 +224,122 @@ and the actor replies to the endpoint when the response is ready. The ask reques
...
 
+@@@ div { .group-scala }
+
 @@snip [Producers.scala]($code$/scala/docs/camel/Producers.scala) { #RouteResponse }
 
+@@@ 
+
+@@@ div { .group-java }
+
+@@snip [ResponseReceiver.java]($code$/java/jdocs/camel/ResponseReceiver.java) { #RouteResponse }
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: Three code snippets in different files, so we cannot use tabs to separate Java/Scala, but have to enclose them in `@@@ div { .group-java }`

File: akka-docs/src/main/paradox/scala/camel.md [hunk 6]
```diff
@@ -333,13 +437,30 @@ akka://some-system/user/myconsumer?autoAck=false&replyTimeout=100+millis
...
 ends at the target actor.
 
 @@snip [CustomRoute.scala]($code$/scala/docs/camel/CustomRoute.scala) { #CustomRoute }
 
-When a message is received on the jetty endpoint, it is routed to the Responder actor, which in return replies back to the client of
+@@@
+
+@@@ div { .group-java }
+
+@@snip [Responder.java]($code$/java/jdocs/camel/Responder.java) { #CustomRoute }
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: Same as lines from 279~, three code snippets in different files, so we cannot use tabs to separate Java/Scala, but have to enclose them in `@@@ div { .group-java }`

File: akka-docs/src/main/paradox/scala/camel.md [hunk 7]
```diff
@@ -333,13 +437,30 @@ akka://some-system/user/myconsumer?autoAck=false&replyTimeout=100+millis
 ```
 
 In the following example, a custom route to an actor is created, using the
-actor's path. the Akka camel package contains an implicit `toActorRouteDefinition` that allows for a route to
+actor's path. 
+
+@@@ div { .group-scala }
+
+The Akka camel package contains an implicit `toActorRouteDefinition` that allows for a route to
```
Comment (richardimaoka) [inline akka-docs/src/main/paradox/scala/camel.md]: implicit `toActorRouteDefinition` relevant to Scala only?

**Comment by raboof** [comment]
OK TO TEST

**Comment by raboof** [comment]
paradox doesn't seem to like the 'extref' to 'ask'

File: akka-docs/src/main/paradox/scala/camel.md (hunk 5 above)
Comment (raboof) [inline akka-docs/src/main/paradox/scala/camel.md]: This seems to work:
```
Scala
:   @@snip [Producers.scala]($code$/scala/docs/camel/Producers.scala) { #RouteResponse }

Java
:   @@snip [ResponseReceiver.java]($code$/java/jdocs/camel/ResponseReceiver.java) { #RouteResponse }
    @@snip [Forwarder.java]($code$/java/jdocs/camel/Forwarder.java) { #RouteResponse }
    @@snip [OnRouteResponseTestBase.java]($code$/java/jdocs/camel/OnRouteResponseTestBase.java) { #RouteResponse }
```

It's a bit picky about the exact indentation though.

**Comment by richardimaoka** [comment]
Fixed the paradox error.

File: akka-docs/src/main/paradox/scala/camel.md (hunk 5 above)
Comment (raboof) [inline akka-docs/src/main/paradox/scala/camel.md]: Could you have a look at these ones? after that I think we're good to go!

**Comment by richardimaoka** [comment]
Added a new commit. Thanks @raboof, yes the advised change https://github.com/akka/akka/pull/23456#discussion_r130430897 worked. 
https://richard-akka-doc.firebaseapp.com/java/camel.html

**Comment by raboof** [comment]
Nice!

**Pull request merged**